atomicfu-transformer/src/main/kotlin/kotlinx/atomicfu/transformer/AtomicFUTransformerJS.kt

```diff
@@ -0,0 +1,310 @@
+                }
+            }
+            return true
+        }
+    }
+
+    private fun FunctionCall.inlineAtomicOperation(
+        funcName: String,
+        field: AstNode,
+        args: List<AstNode>,
+        passScope: Boolean
+    ): Boolean {
+        var code: String? = null
+        val f = if (passScope) ("scope" + '.' + (field as PropertyGet).property.toSource()) else field.toSource()
+        when (funcName) {
+            "getAndSet\$atomicfu" -> {
+                val arg = args[0].toSource()
+                code = "(function(scope) {var oldValue = $f; $f = $arg; return oldValue;})"
+            }
+            "compareAndSet\$atomicfu" -> {
+                val expected = args[0].toSource()
+                val updated = args[1].toSource()
+                code =
+                    "(function(scope) {return $f === $expected ? function() { $f = $updated; return true }() : false})"
+            }
+            "getAndIncrement\$atomicfu" -> {
+                code = "(function(scope) {return $f++;})"
+            }
+
+            "getAndIncrement\$atomicfu\$long" -> {
+                code = "(function(scope) {var oldValue = $f; $f = $f.inc(); return oldValue;})"
+            }
+
+            "getAndDecrement\$atomicfu" -> {
+                code = "(function(scope) {return $f--;})"
+            }
+
+            "getAndDecrement\$atomicfu\$long" -> {
+                code = "(function(scope) {var oldValue = $f; $f = $f.dec(); return oldValue;})"
+            }
+
+            "getAndAdd\$atomicfu" -> {
+                val arg = args[0].toSource()
+                code = "(function(scope) {var oldValue = $f; $f += $arg; return oldValue;})"
+            }
+
+            "getAndAdd\$atomicfu\$long" -> {
+                val arg = args[0].toSource()
+                code = "(function(scope) {var oldValue = $f; $f = $f.add($arg); return oldValue;})"
+            }
+
+            "addAndGet\$atomicfu" -> {
+                val arg = args[0].toSource()
+                code = "(function(scope) {$f += $arg; return $f;})"
+            }
+
+            "addAndGet\$atomicfu\$long" -> {
+                val arg = args[0].toSource()
+                code = "(function(scope) {$f = $f.add($arg); return $f;})"
+            }
+
+            "incrementAndGet\$atomicfu" -> {
+                code = "(function(scope) {return ++$f;})"
+            }
+
+            "incrementAndGet\$atomicfu\$long" -> {
+                code = "(function(scope) {return $f = $f.inc();})"
+            }
+
+            "decrementAndGet\$atomicfu" -> {
+                code = "(function(scope) {return --$f;})"
+            }
+
+            "decrementAndGet\$atomicfu\$long" -> {
+                code = "(function(scope) {return $f = $f.dec();})"
+            }
+        }
+        if (code != null) {
+            this.getNode(code)
+            return true
+        }
+        return false
+    }
+
+    private fun FunctionCall.getNode(code: String) {
```

Comment:
Why it is called `getNode` when it does not even return anything? Rename, please.